Unity/NativeObject.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Specialized;
+
+namespace Unity
+{
+    public unsafe class NativeObject : IDisposable
+    {
+        NativeObjectV1* nativeObject;
```

Comment:
This should mirror the implementation of `TypeTree` with an `INativeObjectImpl` interface instead of embedding the structure directly.

Reply:
I wasn't sure if there would ever be more then one version of NativeObject, so i wasn't sure if it should mirror TypeTree?

Reply:
NativeObject has changed a few times across Unity versions, but if I remember correctly the data that changed is located after the bit fields, so it's not data that gets used in the dumper. That said, it's probably better to have a system in place to account for future changes (and just for consistency with the other code).

On that note, I wonder if it might be better to start naming things like `TypeTree.V2019_1` instead of `TypeTree.V2`. That way we don't need to go back and rename a bunch of types if we accidentally miss a revision of the type in between.

Reply:
So, with the TypeTree example, they'd be named TypeTree.V1, TypeTree.V2019_1, TypeTree.V2019_3? Assmuing we support Unity 3.4 (the earliest version of unity still on the download archive, would the TypeTree.V1 be renamed to TypeTree.V3_4?

Reply:
Yeah, that sounds about right. There aren't any plans to support anything older than Unity 3.4 anyway, so naming the `V1` struct `V3_4` sounds fine to me.